gsl/gsl_util

```diff
@@ -135,30 +135,30 @@ inline T narrow(U u)
 // at() - Bounds-checked way of accessing static arrays, std::array, std::vector
 //
 template <class T, size_t N>
-constexpr T& at(T (&arr)[N], size_t index)
+constexpr T& at(T (&arr)[N], std::ptrdiff_t index)
 {
-    Expects(index < N);
-    return arr[index];
+    Expects(index >= 0 && index < narrow_cast<decltype(index)>(N));
+    return arr[static_cast<size_t>(index)];
```

Comment:
Now that I look at it, I probably should have used decltype(N) for size_t instead of using size_t directly for the same reasons